### DISCUSSION TITLE
A Fix For Pin To Start On WIndows 10 Start Menu (OS Build: 22621.1413+) (Use It At Your Own Risk)

### DISCUSSION THREAD

**Discussion by milestprower92** [valinet/ExplorerPatcher#2230] [Show and tell]
**Causes Of The Missing UWP Apps Issue #1108 and Broken Pin To Start #1679 On Windows 10 Style Start Menu:**

StartMenu.dll Is The New Data Base That sends the Orders to Pin Stuff Only To The Windows 11 Start Menu.

On 21H2 StartMenu.dll didn't exist and that's why pin to start was working until build 22621.1344 and higher.
All This Happened Because Microsoft Introduced StartMenu.dll, Causing The Non UWP Apps Issue On The Windows 10 Start Menu and When you tried to Use The Inbox Version Of StartTileData.dll And AppResolver.dll, Resulted that UWP Apps Returned To The Older Start Menu But Pining Was Broken, All Thanks To StartMenu.dll That Sended The Commands To The WIndows 11 Start Menu Causing All of Those Issues.

Tutorial of How To fix Pin To Start (Use it at your own risk):

1. Press The "Windows key" And "R"
2. Paste This: C:\Windows\SystemApps\MicrosoftWindows.Client.Core_cw5n1h2txyewy
and Press Enter
5. Search "StartMenu.dll"
6. Take Ownership of The File
7. Remane it To: "Startmenu11.dll
8. Restart Explorer

And After That, Pin To Start Will be Fixed on Start and search,
This Solution Will Work Until @valinet Finds a Complete Solution To Explorer Patcher

Enjoy! :)

**Comment by valinet** [comment]
Thanks the the tip.

Although my recommendation is to not use this; let us figure out a solution that is manageable by the end user, not one that requires users to replace system files. A bit of investigation will tell how that DLL is used by which processes and we can operate a redirection just on those by hooking, instead of modifying the system files which get reverted with each Windows update. Just a bit of time, @Amrsatrio is looking on it as well.

**Reply by milestprower92** [reply]
Ok, but it will be useful for some users that want the fixed pin to start before you find a solution,
like on #1108 that they used the inbox version of starttiledata.dll and appresolver.dll or renaming dxgi.dll
